basic.go

```diff
@@ -21,15 +32,486 @@ func (cb *Cinnabot) Echo(msg *message) {
+func (cb *Cinnabot) Capitalize(msg *message) {
+	cb.SendTextMessage(int(msg.Chat.ID), strings.ToUpper(strings.Join(msg.Args, " ")))
+}
+
+//Start initializes the bot
+func (cb *Cinnabot) Start(msg *message) {
+	text := "Hello there " + msg.From.FirstName + "!\n\n" +
+		"Im Cinnabot🤖. I am made by my owners to serve the residents of Cinnamon college!\n" +
+		"Im always here to /help if you need it!"
+
+	cb.SendTextMessage(int(msg.Chat.ID), text)
+}
+
+// Help gives a list of handles that the user may call along with a description of them
+func (cb *Cinnabot) Help(msg *message) {
+	if len(msg.Args) > 0 {
+
+		if msg.Args[0] == "spaces" {
+			text :=
+				"To use the '/spaces' command, type one of the following:\n" +
+					"'/spaces' : to view all bookings for today\n'/spaces now' : to view bookings active at this very moment\n" +
+					"'/spaces week' : to view all bookings for this week\n'/spaces dd/mm(/yy)' : to view all bookings on a specific day\n" +
+					"'/spaces dd/mm(/yy) dd/mm(/yy)' : to view all bookings in a specific range of dates"
+			cb.SendTextMessage(int(msg.Chat.ID), text)
+			return
+
+		} else if msg.Args[0] == "cbs" {
+			text :=
+				"/subscribe <tag>: subscribe to a tag\n" +
+					"/unsubscribe <tag>: unsubscribe from a tag\n" +
+					"/broadcast <tag>: broadcast to a tag [admin]\n" +
+					"Alternatively you can just type:\n" +
+					"/subscribe for a button list\n" +
+					"/unsubscribe for a button list\n"
+			cb.SendTextMessage(int(msg.Chat.ID), text)
+			return
+		} else if msg.Args[0] == "resources" {
+			text :=
+				"/resources <tag>: searches resources for a specific tag\n" +
+					"/resources: returns all tags"
+			cb.SendTextMessage(int(msg.Chat.ID), text)
+			return
+		} else if msg.Args[0] == "publicbus" {
+			text :=
+				"/publicbus : publicbus\n" +
+					"Sending your location (ignore the buttons) after running the above command will allow to get bus timings for bus stops around any location."
+			cb.SendTextMessage(int(msg.Chat.ID), text)
+			return
+		}
+	}
+	text :=
+		"Here are a list of functions to get you started 🤸 \n" +
+			"/about: to find out more about me\n" +
+			"/cbs: cinnamon broadcast system\n" +
+			"/publicbus: public bus timings for bus stops around your location\n" +
+			"/nusbus: nus bus timings for bus stops around your location\n" +
+			"/weather: 2h weather forecast\n" +
+			"/resources: list of important resources!\n" +
+			"/spaces: list of space bookings\n" +
+			"/feedback: to give feedback\n\n" +
+			"_*My creator actually snuck in a few more functions🕺 *_\n" +
+			"Try using /help <func name> to see what I can _really_ do"
+	cb.SendTextMessage(int(msg.Chat.ID), text)
 }
 
 // About returns a link to Cinnabot's source code.
 func (cb *Cinnabot) About(msg *message) {
-	cb.SendTextMessage(msg.From.ID, "Touch me: https://github.com/varunpatro/Cinnabot")
+	cb.SendTextMessage(int(msg.Chat.ID), "Touch me: https://github.com/varunpatro/Cinnabot")
```

Comment:
You should add yourself here too! Say it was developed by you too (before showing the link)